From 3c11a25943a2d70a9a93ddf4ae627a2be9b4a81f Mon Sep 17 00:00:00 2001
From: Tammo Jan Dijkema <dijkema@astron.nl>
Date: Thu, 16 May 2024 13:38:46 +0200
Subject: [PATCH] Add missing include in test

Fixes #1348
---
 tables/Dysco/tests/testdyscostman.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tables/Dysco/tests/testdyscostman.cc b/tables/Dysco/tests/testdyscostman.cc
index 69da97cf078..ae52c0364c6 100644
--- a/tables/Dysco/tests/testdyscostman.cc
+++ b/tables/Dysco/tests/testdyscostman.cc
@@ -1,5 +1,6 @@
 #include <boost/test/unit_test.hpp>
 #include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/directory.hpp>
 
 #include <casacore/tables/Tables/ArrayColumn.h>
 #include <casacore/tables/Tables/Table.h>
