diff --git a/CMakeLists.txt b/CMakeLists.txt
index 774aa19..16a0cd0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -533,6 +533,10 @@ else( )
     set (LIB_SOVERSION "${PROJECT_SOVERSION}")
 endif( )
 
+# conda-build hack
+add_definitions(-DCONDA_CASA_ROOT="${CONDA_CASA_ROOT}")
+# end conda-build hack
+
 # Add the modules to be built.
 add_subdirectory (build-tools)
 foreach (module ${_modules})
diff --git a/casa/System/Aipsrc.cc b/casa/System/Aipsrc.cc
index 1c18339..93f9794 100644
--- a/casa/System/Aipsrc.cc
+++ b/casa/System/Aipsrc.cc
@@ -200,9 +200,9 @@ void Aipsrc::fillAips() {
   } else {
     aipsPath = extAipsPath;
   }
-  // Set the path to home if not defined in any way.
+  // conda-build hack: default is known; hack to fix zero-pad issues on OSX
   if (aipsPath.empty()) {
-    setAipsPath(uhome);
+    setAipsPath(String(String(CONDA_CASA_ROOT).chars()));
     aipsPath = extAipsPath;
   }
   Int n = std::count(aipsPath.begin(), aipsPath.end(), ' ') +
diff --git a/measures/Measures/MeasIERS.cc b/measures/Measures/MeasIERS.cc
index fc11b35..8e8d09b 100644
--- a/measures/Measures/MeasIERS.cc
+++ b/measures/Measures/MeasIERS.cc
@@ -405,7 +405,8 @@ Bool MeasIERS::findTab(Table &tab, const Table *tabin, const String &rc, const S
               }
             }
             if (!found) {
-              String casadata = String(CASADATA);
+              // Conda NUL-padding workaround:
+              String casadata = String(String(CASADATA).chars());
               ReplaceAllInPlace(casadata, "%CASAROOT%", Aipsrc::aipsRoot());
               ReplaceAllInPlace(casadata, "%CASAHOME%", Aipsrc::aipsHome());
               Path cdatapath(casadata);
diff --git a/measures/Measures/MeasTable.cc b/measures/Measures/MeasTable.cc
index e35fda5..7e0e36c 100644
--- a/measures/Measures/MeasTable.cc
+++ b/measures/Measures/MeasTable.cc
@@ -2758,7 +2758,8 @@ Bool MeasTable::AntennaResponsesPath(String &antRespPath, const String &nam) {
         isValid = Table::isReadable(absPathName);
       }
       if (!isValid) {
-        String casadata = String(CASADATA);
+        // Conda NUL-padding workaround:
+        String casadata = String(String(CASADATA).chars());
         ReplaceAllInPlace(casadata, "%CASAROOT%", Aipsrc::aipsRoot());
         ReplaceAllInPlace(casadata, "%CASAHOME%", Aipsrc::aipsHome());
         Path lPath(casadata + "/" + antRespPath);
