--- a/casacore/CMakeLists.txt
+++ b/casacore/CMakeLists.txt
@@ -314,6 +314,10 @@
     add_definitions(-DUSE_STACKTRACE)
 endif(USE_STACKTRACE)
 
+# conda-build hack
+add_definitions(-DCONDA_CASA_ROOT="${CONDA_CASA_ROOT}")
+# end conda-build hack
+
 # Add the modules to be built.
 add_subdirectory (build-tools)
 foreach (module ${_modules})
--- a/casacore/casa/System/Aipsrc.cc
+++ b/casacore/casa/System/Aipsrc.cc
@@ -215,9 +215,9 @@
     } else { 
       aipsPath = extAipsPath;
     }
-    // Set the path to home if not defined in any way.
+    // conda-build hack: default is known; hack to fix zero-pad issues on OSX
     if (aipsPath.empty()) {
-      setAipsPath(uhome);
+      setAipsPath(String(String(CONDA_CASA_ROOT).chars()));
       aipsPath = extAipsPath;
     }
     Int n = aipsPath.freq(' ') + aipsPath.freq('	') + 4;
@@ -422,7 +422,7 @@
     if (find(x, String("user.aipsdir"), String("/aips++"))) {
       home = x;
     } else {
-      home = uhome + x;
+      home = String(String(CONDA_CASA_ROOT).chars());
     }
   }
 }
--- a/casacore/measures/Measures/MeasIERS.cc
+++ b/casacore/measures/Measures/MeasIERS.cc
@@ -422,7 +422,8 @@
           }
         }
         if (!found) {
-          String casadata=String(CASADATA);
+	  // Conda NUL-padding workaround:                                                                        
+          String casadata=String(String(CASADATA).chars());
           casadata.gsub("%CASAROOT%", Aipsrc::aipsRoot());
           casadata.gsub("%CASAHOME%", Aipsrc::aipsHome());
           Path cdatapath(casadata);
--- a/casacore/measures/Measures/MeasTable.cc
+++ b/casacore/measures/Measures/MeasTable.cc
@@ -2988,7 +2988,8 @@
 	isValid = Table::isReadable(absPathName);
       }
       if(!isValid){
-        String casadata=String(CASADATA);
+	// Conda NUL-padding workaround:                                                                        
+        String casadata=String(String(CASADATA).chars());
         casadata.gsub("%CASAROOT%", Aipsrc::aipsRoot());
         casadata.gsub("%CASAHOME%", Aipsrc::aipsHome());
         Path lPath(casadata + "/" + antRespPath);
