diff --git a/CMakeLists.txt b/CMakeLists.txt
index c14499a..38f8562 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -527,6 +527,10 @@ else( )
     set (LIB_SOVERSION "${PROJECT_SOVERSION}")
 endif( )
 
+# conda-build hack
+add_definitions(-DCONDA_CASA_ROOT="${CONDA_CASA_ROOT}")
+# end conda-build hack
+
 # Add the modules to be built.
 add_subdirectory (build-tools)
 foreach (module ${_modules})
diff --git a/casa/System/Aipsrc.cc b/casa/System/Aipsrc.cc
index d22ec93..fb1b532 100644
--- a/casa/System/Aipsrc.cc
+++ b/casa/System/Aipsrc.cc
@@ -220,9 +220,9 @@ void Aipsrc::fillAips() {
   } else {
     aipsPath = extAipsPath;
   }
-  // Set the path to home if not defined in any way.
+  // conda-build hack: default is known; hack to fix zero-pad issues on OSX
   if (aipsPath.empty()) {
-    setAipsPath(uhome);
+    setAipsPath(String(String(CONDA_CASA_ROOT).chars()));
     aipsPath = extAipsPath;
   }
   Int n = aipsPath.freq(' ') + aipsPath.freq('	') + 4;
diff --git a/measures/Measures/MeasIERS.cc b/measures/Measures/MeasIERS.cc
index 6c82900..8062e01 100644
--- a/measures/Measures/MeasIERS.cc
+++ b/measures/Measures/MeasIERS.cc
@@ -464,7 +464,8 @@ Bool MeasIERS::findTab(Table& tab, const Table *tabin, const String &rc,
               }
             }
             if (!found) {
-              String casadata=String(CASADATA);
+              // Conda NUL-padding workaround:
+              String casadata=String(String(CASADATA).chars());
               casadata.gsub("%CASAROOT%", Aipsrc::aipsRoot());
               casadata.gsub("%CASAHOME%", Aipsrc::aipsHome());
               Path cdatapath(casadata);
diff --git a/measures/Measures/MeasTable.cc b/measures/Measures/MeasTable.cc
index 42e64a5..7d3df47 100644
--- a/measures/Measures/MeasTable.cc
+++ b/measures/Measures/MeasTable.cc
@@ -2939,7 +2939,8 @@ Bool MeasTable::AntennaResponsesPath(String &antRespPath, const String &nam) {
 	isValid = Table::isReadable(absPathName);
       }
       if(!isValid){
-        String casadata=String(CASADATA);
+        // Conda NUL-padding workaround:
+        String casadata=String(String(CASADATA).chars());
         casadata.gsub("%CASAROOT%", Aipsrc::aipsRoot());
         casadata.gsub("%CASAHOME%", Aipsrc::aipsHome());
         Path lPath(casadata + "/" + antRespPath);
