diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9601f85..f26ef96 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,9 @@ include(CheckCCompilerFlag)
 include(CheckFunctionExists)
 include(CheckCXXSourceCompiles)
 
+# conda-forge patch:
+list(FILTER CMAKE_SYSTEM_PREFIX_PATH EXCLUDE REGEX build_env)
+
 # Fixes warnings on cmake 3.x+ about missing extensions
 if (POLICY CMP0115)
     cmake_policy (SET CMP0115 NEW)
