diff --git a/CMakeLists.txt b/CMakeLists.txt
index be46fd5..a962877 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,9 @@ include(CheckFunctionExists)
 include(CheckCXXSourceCompiles)
 include(cmake/PCHSupport.cmake)
 
+# conda-forge patch:
+list(FILTER CMAKE_SYSTEM_PREFIX_PATH EXCLUDE REGEX build_env)
+
 # Fixes warnings on cmake 3.x+ about missing extensions
 if (POLICY CMP0115)
     cmake_policy (SET CMP0115 NEW)
