diff --git a/CMakeLists.txt b/CMakeLists.txt
index a962877..2514e61 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -329,7 +329,7 @@ if (USE_READLINE)
 endif (USE_READLINE)
 find_package (SOFA)
 if (USE_ADIOS2)
-    find_package (ADIOS2 2.6.0 CONFIG REQUIRED)
+    find_package (ADIOS2 2.6.0 REQUIRED COMPONENTS C)
     if(NOT USE_MPI)
         set(CASACORE_ADIOS_LIBRARY adios2::cxx11)
     elseif (NOT ADIOS2_HAVE_MPI)
